sunpy/util/tests/test_metadata.py

```diff
@@ -0,0 +1,224 @@
+    'same_diff_keys_data' in a MetaDict
+    """
+    return [['labrador', 'americas'],
+            ['Norwegian', 'NORWAY'],
+            ['BALTIC', 'europe'],
+            ['LaPteV', 'russia'],
+            ['SCOTIA', 's.ocean'],
+            ['Mawson', 'E.Antartica']]
+
+
+@pytest.fixture
+def std_metadict(std_test_data):
+    return MetaDict(std_test_data)
+
+
+def test_init_with_lst_of_lsts(std_test_data):
+    check(MetaDict(std_test_data), std_test_data)
+
+
+def test_init_with_tuple_of_tuples(std_test_data):
+    in_tuple = tuple(tuple(kv_pair) for kv_pair in std_test_data)
+    check(MetaDict(in_tuple), std_test_data)
+
+
+def test_init_with_dict(std_test_data):
+    # Using traditional dictionary, order of insertion is *not* preserved
+    check_contents(MetaDict(pairs_to_dict(std_test_data)), std_test_data)
+
+
+def test_init_with_metadict(same_diff_keys_data):
+    check(MetaDict(MetaDict(same_diff_keys_data)), same_diff_keys_data)
+
+
+def test_init_with_illegal_arg():
+    with pytest.raises(TypeError):
+        MetaDict(set(('a', 'b', 'c', 'd')))
+
+
+def test_membership_op(std_metadict):
+    assert 'LABRADOR' in std_metadict
+    assert 'labrador' in std_metadict
+    assert 'baLtiC' in std_metadict
+
+    assert 'pechora' not in std_metadict
+
+
+def test_getitem_op(std_metadict):
+    assert std_metadict['norwegian'] == 'Europe'
+    assert std_metadict['NORWEGIAN'] != 'europe'
+    assert std_metadict['laptev'] == 'arctic'
+
+    assert std_metadict['Norwegian'] != 'europe'
+
+    with pytest.raises(KeyError):
+        std_metadict['key-not-exists']
+
+
+def test_get_method(std_metadict):
+    assert std_metadict.get('baltic') == 'europe'
+    assert std_metadict.get('atlantic') is None
+
+    default_value = 'Unknown'
+    assert std_metadict.get('sargasso', default=default_value) == default_value
+
+
+def test_setitem_op_existing(std_metadict):
+    len_before = len(std_metadict)
+    std_metadict['NORWEGIAN'] = 'Scandinavia'
+    assert len(std_metadict) == len_before
+
+    assert std_metadict['norwegian'] == 'Scandinavia'
+    assert std_metadict['NoRwEgIaN'] == 'Scandinavia'
+
+
+def test_setitem_op_new(std_metadict):
+    len_before = len(std_metadict)
+    std_metadict['Irish'] = 'N.Europe'
+    assert len(std_metadict) == len_before + 1
+
+    assert std_metadict['Irish'] == 'N.Europe'
+    assert std_metadict['irish'] == 'N.Europe'
+    assert std_metadict['IRISH'] == 'N.Europe'
+
+    assert std_metadict['IrisH'] != 'n.europe'
+
+
+def test_setdefault_method(std_metadict):
+    std_metadict.setdefault('poseidon', 'NOT-FOUND')
+    assert std_metadict.get('Poseidon') == 'NOT-FOUND'
+    assert std_metadict['pOSeidon'] == 'NOT-FOUND'
+
+    # Should only impact key 'poseidon'
+    assert std_metadict.get('Wandel') is None
+
+
+def test_has_key_method(std_metadict):
+
+    # suppress pep8/flake8 'W601 .has_key() is deprecated, ...' warnings
+    # as MetaDict explicitly supports the 'has_key()' method
```

Comment:
This is interesting. Do we want to deprecate `has_key` on our `MetaDict` as well? @Alex-Ian-Hamilton ??